examples/bodyparser.rs

```diff
@@ -1,18 +1,97 @@
 extern crate iron;
 extern crate bodyparser;
 extern crate persistent;
-extern crate rustc_serialize;
+extern crate serde;
 
 use persistent::Read;
 use iron::status;
 use iron::prelude::*;
+use serde::{Deserialize, Deserializer};
+use serde::de::{MapVisitor, Visitor};
 
-#[derive(Debug, Clone, RustcDecodable)]
+#[derive(Debug, Clone)]
 struct MyStructure {
     a: String,
     b: Option<String>,
 }
 
+impl Deserialize for MyStructure {
```

Comment:
Can you add a comment here explaining why we are doing this and maybe linking to how not to have to do this?